### PR TITLE
Reset signals caught by clickhouse-client if a pager is in use

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Reset signals that is catched by clickhouse-client if pager is in use

We need to reset signals that had been installed in the setupSignalHandler() (SIGINT/SIGQUIT), like other such tools dies (i.e. less and open file in editor) since terminal will send signals to both processes and so signals will be delivered to the
clickhouse-client/local as well, which will be terminated when signal will be delivered second time.

So a simple `clickhouse-client --pager less` will be terminated on a second C-C press. And it is really annoying when you are using this way to analyze logs.

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 6c5f1c22acd62f42c3c766905d00725c4a2a00b2 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55193/6c5f1c22acd62f42c3c766905d00725c4a2a00b2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6364319003">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55193/6c5f1c22acd62f42c3c766905d00725c4a2a00b2/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by azat** [comment]
Can someone take a look please?

File: src/Client/ClientBase.cpp [hunk 1]
```diff
@@ -564,7 +564,16 @@ try
...
         {
             if (SIG_ERR == signal(SIGPIPE, SIG_IGN))
-                throwFromErrno("Cannot set signal handler.", ErrorCodes::CANNOT_SET_SIGNAL_HANDLER);
+                throwFromErrno("Cannot set signal handler for SIGPIPE.", ErrorCodes::CANNOT_SET_SIGNAL_HANDLER);
+            /// We need to reset signals that had been installed in the
+            /// setupSignalHandler() since terminal will send signals to both
+            /// processes and so signals will be delivered to the
+            /// clickhouse-client/local as well, which will be terminated when
+            /// signal will be delivered second time.
+            if (SIG_ERR == signal(SIGINT, SIG_IGN))
```
Comment (Algunenano) [inline src/Client/ClientBase.cpp:573]: If you ignore signals, how can you cancel a query that's being run?

For example, if I do `--pager less` send a neverending query (`Select * from system.numbers`) and go to the end (`G command`) now I can't cancel the query from the  CLI. Same if you use `less +F`, you stop being able to stop the query at all as ctrl+c does nothing.

Comment (Algunenano) [reply src/Client/ClientBase.cpp:573]: Makes sense. With `output_format_pretty_max_rows` disabled it seems to be working fine. I'm not sure if I'd expect  Ctrl+C to also stop the source but I guess not, and doing it on pager exit seems the most reasonable way forward.

**Review by Algunenano — APPROVED** [review]

**Comment by Algunenano** [comment]
* `Integration tests (asan) [1/6] — fail: 1, passed: 396` -> `test_dictionaries_redis/test_long.py::test_redis_dict_long` -> Timeout when connecting to redis (5 seconds). The test seems quite  flaky in the [last couple of days](https://play.clickhouse.com/play?user=play#U0VMRUNUCiAgY2hlY2tfbmFtZSwKICB0ZXN0X25hbWUsCiAgdG9TdGFydE9mRGF5KGNoZWNrX3N0YXJ0X3RpbWUpIGFzIHQsCiAgY291bnQoKSBhcyBydW5zLAogIGNvbmNhdChyb3VuZCgxMDAgKiAoY291bnRJZih0ZXN0X3N0YXR1cyAhPSAnT0snIEFORCB0ZXN0X3N0YXR1cyAhPSAnU0tJUFBFRCcpIEFTIGYpIC8gcnVucywgMSk6OlN0cmluZywgJyUnKSBhcyBmYWlsdXJlX3BlcmNlbnRhZ2UsCiAgYW55SWYocmVwb3J0X3VybCwgdGVzdF9zdGF0dXMgIT0gJ09LJyBBTkQgdGVzdF9zdGF0dXMgIT0gJ1NLSVBQRUQnKSBhcyByZXBvcnQKRlJPTSBjaGVja3MKV0hFUkUKICAgIHRlc3RfbmFtZSBMSUtFICd0ZXN0X2RpY3Rpb25hcmllc19yZWRpcy90ZXN0X2xvbmcucHk6OnRlc3RfcmVkaXNfZGljdF9sb25nJScKICAgIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIGNoZWNrX3N0YXJ0X3RpbWUgPiB0b2RheSgpIC0gaW50ZXJ2YWwgMzAgZGF5CkdST1VQIEJZIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdApIQVZJTkcgZiA+IDAKT1JERVIgYnkgY2hlY2tfbmFtZSwgdGVzdF9uYW1lLCB0)

**Pull request merged**